### PR TITLE
Lookup tokens by username

### PR DESCRIPTION
**Pull request by sebinjohn**

### CODE REVIEW

**Comment by sebinjohn** [comment]
Can I help in anyway to progress this PR?

File: artifactory/services/security.go [hunk 1]
```diff
@@ -124,6 +125,22 @@ func (ss *SecurityService) RevokeToken(params RevokeTokenParams) (string, error)
...
+// LookupTokenID looks up a token by username and return token id
+func (ss *SecurityService) LookupTokenID(username string) (string, error) {
+	tokenResponseData, err := ss.GetTokens()
+	if err != nil {
+		return "", err
+	}
+
+	for _, token := range tokenResponseData.Tokens {
+		if strings.HasSuffix(token.Subject, "/"+username) {
+			return token.TokenId, nil
```
[the remark this answers is not in the document]
Comment (sebinjohn) [reply artifactory/services/security.go]: Thanks. Added support for looking up all the tokens by a user.

File: artifactory/services/security.go [hunk 2]
```diff
@@ -124,6 +125,23 @@ func (ss *SecurityService) RevokeToken(params RevokeTokenParams) (string, error)
 	return string(body), err
 }
 
+// LookupTokenID looks up a token by username and return token id
+func (ss *SecurityService) LookupTokenID(username string) ([]string, error) {
```
Comment (eyalbe4) [inline artifactory/services/security.go]: Let's rename the function name to more accurately reflect what is does. I suggest renaming it to *GetUserTokens*.
Let's also change the function's description and references in the tests accordingly.
In addition, let's add the new API to the README.

**Comment by sebinjohn** [comment]
@eyalbe4 I have addressed the comments. Can you have another look?

**Comment by eyalbe4** [comment]
@sebinjohn,
*getUserTokensTest* fails, when the the user already has tokens. which have been created before.
I suggest modifying the test, so that it fails if the created token (the expected one) is not included in the list of tokens fetched.

**Comment by sebinjohn** [comment]
> @sebinjohn,
> _getUserTokensTest_ fails, when the the user already has tokens. which have been created before.
> I suggest modifying the test, so that it fails if the created token (the expected one) is not included in the list of tokens fetched.

Done

**Comment by eyalbe4** [comment]
@sebinjohn,
There still sees to be an issue with the test logic.
The *expectedTokenID* is not necessarily the token the test has just created. More than that, it can even be a token for a different user.
I recommend using the token returned by the *createToken()* API, to be the *expectedTokenID*. There's probably no need to call the *testsSecurityService.GetTokens()* API in this test.

**Comment by eyalbe4** [comment]
@sebinjohn,
Are you planning to continue working on this?

**Comment by eyalbe4** [comment]
I'm closing this one for now @@sebinjohn.

**Pull request closed without merging**